### PR TITLE
Docker Build Pipeline with Tags

### PR DESCRIPTION
**Pull request by gimse**
Making Github Action that builds docker image with "version number" and "latest". 
For example ghcr.io/protomaps/go-pmtiles:v0.0.7 & ghcr.io/gimse/go-pmtiles:latest

It still builds ghcr.io/protomaps/go-pmtiles:main on push to main. 

It also run the pipeline on pull requests, but do not push to the registry.

### CODE REVIEW

**Comment by gimse** [comment]
Example on how it looks in to container registry : 

image

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
-    name: Push Docker image to GitHub Packages as with Tag 'latest'
+  docker-build:
     runs-on: ubuntu-latest
-    if: github.ref == 'refs/heads/main'
     steps:
-      - name: Check out the repo
-        uses: actions/checkout@v2
-      - name: Push to GitHub Packages
-        uses: docker/build-push-action@v1
+      - uses: docker/setup-qemu-action@v3
```
[the remark this answers is not in the document]
Comment (gimse) [reply .github/workflows/publish.yml]: You are correct @bdon! I have removed the lines now. 

I originally used steps to build for different platform (linux/amd64, linux/arm64, linux/arm/v7), but decided to remove the feature.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
 
-on: [push]
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
```
Comment (bdon) [inline .github/workflows/publish.yml:10]: Is this `pull_request` key necessary - shouldn't it be canceled out by `push: ${{ github.event_name != 'pull_request' }}` on line 32?

**Comment by bdon** [comment]
Tagging @fraank original author of the GitHub action - does this PR work for Docker container usage?

**Comment by gimse** [comment]
Thanks :)

**Pull request merged**